retail/retail-java-applications/terraform/test/README.md

```diff
@@ -0,0 +1,117 @@
+# Automated Test For Terraform Modules
+
+Use Go and Terratest to test the following modules.
+- Pub/Sub
+
+## Install Terraform and Go
+
+If you are using the [Google Cloud Shell](https://cloud.google.com/shell/docs/how-cloud-shell-works), both Terraform and Go are installed.
+
+Follow the instructions to install [Terraform cli](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/gcp-get-started) and [Go](https://golang.org/doc/install).
+
+This repo has been tested on Terraform version `0.14.5`, Go version `1.16` and Google provider version `3.48.0`
+
+## Terraform Folder Structure
+
+
+The Terraform folder structure can be found below. 
+
+Note that the `terraform` folder is inside `dataflow-sample-applications/retail/retail-java-applications` folder.
+
+```
+terraform
+├── examples
+│   └── pubsub
+│       ├── main.tf
+│       ├── terraform.tfvars
+│       └── variables.tf
+├── modules
+│   └── pubsub
+│       ├── main.tf
+│       └── variables.tf
+├── test
+│   ├── pubsub
+│   │   ├── go.mod
+│   │   ├── go.sum
+│   │   └── pubsub_test.go
+│   └── README.md
+├── README.MD
+├── terraform.tfvars
+├── main.tf
+└── variables.tf
+```
+
+The `examples` folder contains Terraform scripts that serve as living documentation about how to use the module. They are also used by automated tests for execution.
+
+The `modules` folder contains the scripts that form the basis of the module.
+
+The `test` folder contains the test code.
+
+The `terraform` folder contains Terraform scripts which deploy required resources for `retail-java-application` using corresponding Terraform modules.
+
+## Resources Being Tested
+
+### Pub/Sub
+
+In this test, four topics and three subscriptions will be created in Pub/Sub with prefixes described below followed by a randomised string that consists of 5 lowercase letters. 
+
+For example, the topic name could be `Clickstream-inbound-kpocn` while its corresponding subscription name could be `Clickstream-inbound-sub-wtutp` in the automated test.
+
+This could prevent potential failures for subsequent executions when creating and destroying topics and subscriptions using the same names repeatedly.
```

Comment:
Make "creating and destroying topics and subscriptions" link to "https://cloud.google.com/pubsub/docs/admin#deleting_a_topic"